luigi/contrib/s3.py

```diff
@@ -403,110 +342,109 @@ def copy(self, source_path, destination_path, threads=100, start_time=None, end_
+    is_dir = isdir  # compatibility with old version.
 
-            # This will raise any exceptions in any of the copy threads
-            for result in results:
-                result.get()
+    def mkdir(self, path, parents=True, raise_if_exists=False):
+        if raise_if_exists and self.isdir(path):
+            raise FileAlreadyExists()
 
-            # finish the copy, making the file available in S3
-            mp.complete_upload()
-            return mp.key_name
+        bucket, key = self._path_to_bucket_and_key(path)
+        if self._is_root(key):
+            # isdir raises if the bucket doesn't exist; nothing to do here.
+            return
 
-        except BaseException:
-            logger.info('Error during multipart s3 copy for %s/%s to %s/%s...', src_bucket, src_key, dst_bucket, dst_key)
-            # cancel the copy so we don't get charged for storage consumed by copied parts
-            if mp:
-                mp.cancel_upload()
-            raise
+        path = self._add_path_delimiter(path)
 
-    def move(self, source_path, destination_path, **kwargs):
-        """
-        Rename/move an object from one S3 location to another.
+        if not parents and not self.isdir(os.path.dirname(path)):
+            raise MissingParentDirectory()
 
-        :param kwargs: Keyword arguments are passed to the boto function `copy_key`
-        """
-        self.copy(source_path, destination_path, **kwargs)
-        self.remove(source_path)
+        return self.put_string("", path)
 
     def listdir(self, path, start_time=None, end_time=None, return_key=False):
         """
         Get an iterable with S3 folder contents.
         Iterable contains paths relative to queried path.
-
-        :param start_time: Optional argument to list files with modified dates after start_time
-        :param end_time: Optional argument to list files with modified dates before end_time
-        :param return_key: Optional argument, when set to True will return a boto.s3.key.Key (instead of the filename)
+        :param start_time: Optional argument to list files with modified datetime after start_time
+        :param end_time: Optional argument to list files with modified datetime before end_time
+        :param return_key: Optional argument, when set to True will return boto3's ObjectSummary (instead of the filename)
         """
         (bucket, key) = self._path_to_bucket_and_key(path)
 
         # grab and validate the bucket
-        s3_bucket = self.s3.get_bucket(bucket, validate=True)
+        s3_bucket = self.s3.Bucket(bucket)
 
         key_path = self._add_path_delimiter(key)
         key_path_len = len(key_path)
-        for item in s3_bucket.list(prefix=key_path):
-            last_modified_date = time.strptime(item.last_modified, "%Y-%m-%dT%H:%M:%S.%fZ")
+        for item in s3_bucket.objects.filter(Prefix=key_path):
+            last_modified_date = item.last_modified
```

Comment:
Is `item.last_modified` returned as a string or a date?

Reply:
this is a datetime object
http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Object.last_modified